posts/navarroaxel/18.git-rebase-interactive.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Rewriting the history with Git rebase
+published: true
+description: How to squash commits and keep your branch up-to-date using git rebase 
+tags: 'git, bash, productivity, tooling'
+cover_image: ./assets/git-rebase.jpg
+id: 1949937
+---
+
+As when we work in a new feature or bug fix it is recommended to keep our branch up-to-date with the target branch (`main`, `development`, etc.), we can use the [`git rebase`](https://git-scm.com/docs/git-rebase) command instead of `git merge` in order to keep our Git history cleaner.
+
+The `git rebase` command reapplies our commits on top of another branch, rewriting the Git history. This is not recommended on shared branches (like `main`) because it causes divergence (and therefore conflict solving) and panic on your teammates 😬 since it is sometimes hard to resolve.
```

Comment:
```suggestion
The `git rebase` command reapplies our commits on top of the target branch, rewriting the Git history. This is not recommended on shared branches (like `main`) because it causes divergence (and therefore the need of conflict solving) and panic on your teammates 😬 since it is sometimes hard to resolve.
```